packages/storage/src/indexed-db/stream.ts

```diff
@@ -0,0 +1,24 @@
+import { AnyMessage, JsonValue, Message, MessageType } from '@bufbuild/protobuf';
+import { PenumbraDb, PenumbraStoreNames } from '@penumbra-zone/types';
+import { IDBPCursorWithValue } from 'idb';
+
+export class IdbCursorSource<N extends PenumbraStoreNames, T extends Message<T> = AnyMessage>
+  implements UnderlyingDefaultSource<T>
+{
+  constructor(
+    private cursor: Promise<IDBPCursorWithValue<PenumbraDb, [N], N> | null>,
+    private messageType: MessageType<T>,
+  ) {}
+
+  start(cont: ReadableStreamDefaultController<T>) {
+    // immediately stream as fast as possible, to prevent idb from closing the tx
```

Comment:
We should probably add a bit more documentation here and link to: https://github.com/jakearchibald/idb?tab=readme-ov-file#transaction-lifetime